ibm/resource_ibm_private_dns_glb_pool_test.go

```diff
@@ -0,0 +1,148 @@
+package ibm
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccIBMPrivateDNSGlbPool_Basic(t *testing.T) {
+	var resultprivatedns string
+	name := fmt.Sprintf("testpdnspn%s.com", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckIBMPrivateDNSGlbPoolDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckIBMGlbPoolBasic(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIBMGlbPoolExists("ibm_dns_glb_pool.test-pdns-pool-nw", resultprivatedns),
+				),
+			},
+		},
+	})
+}
+
+func TestAccIBMPrivateDNSGlbPoolImport(t *testing.T) {
+	var resultprivatedns string
+	name := fmt.Sprintf("testpdnszone%s.com", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckIBMPrivateDNSGlbPoolDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckIBMGlbPoolBasic(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIBMGlbPoolExists("ibm_dns_glb_pool.test-pdns-pool-nw", resultprivatedns),
+				),
+			},
+			resource.TestStep{
+				ResourceName:      "ibm_dns_glb_pool.test-pdns-pool-nw",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckIBMGlbPoolBasic(name string) string {
+	return fmt.Sprintf(`
+	data "ibm_resource_group" "rg" {
+		name = "default"
+	}
+	resource "ibm_is_vpc" "test-pdns-glb-pool" {
+		name = "test-pdns-glb-pool"
+		resource_group = data.ibm_resource_group.rg.id
+	}
+	resource "ibm_resource_instance" "test-pdns-glb-pool-instance" {
+		name = "test-pdns-glb-pool-instance"
+		resource_group_id = data.ibm_resource_group.rg.id
+		location = "global"
+		service = "dns-svcs"
+		plan = "standard-dns"
+	}
+	resource "ibm_dns_zone" "test-pdns-glb-pool-zone" {
+		name = "%s"
+		instance_id = ibm_resource_instance.test-pdns-glb-pool-instance.guid
+		description = "testdescription"
+		label = "testlabel"
+	}
+	resource "ibm_dns_glb_pool" "test-pdns-pool-nw" {
```

Comment:
take the input for monitor. Check resource_ibm_cis_origin_pool_test.go file.

Reply:
done